ophyd/_dispatch.py

```diff
@@ -135,6 +174,10 @@ def schedule_utility_task(self, callback, *args, **kwargs):
         'Schedule `callback` with the given args and kwargs in a util thread'
         self._utility_queue.put((callback, args, kwargs))
 
+    def get_thread_context(self, name):
```

Comment:
It may not be worth paying for a function call here when exposing the dict would accomplish the same thing. Is the intent to protect `_thread_contexts` from some evil code that might mutate it, or to make space for the possibility that the process of getting a thread context might grow more complex in the future?

Reply:
It's not likely to get more complicated in the future. Assuming all users are benevolent, I think it comes down to a stylistic choice - which I don't feel strongly enough about to argue about. If you do, I can definitely nuke the function and expose it directly.

Reply:
I'm :+1: on leaving this as-is.

Reply:
Was thinking more about (marginal) performance than style. Fine with leaving it as is.